graphics/freeimage/Portfile

```diff
@@ -30,7 +30,11 @@ distname            FreeImage[strsed ${version} {g/\.//}]
 use_zip             yes
 worksrcdir          FreeImage
 
-patchfiles          patch-c99-fixes.diff
+# https://github.com/WinMerge/freeimage/issues/16
```

Comment:
That's not upstream. Please check https://trac.macports.org/ticket/67489 where I posted the upstream bug URLs. There were multiple proposals for resolving this bug submitted in addition to a different one that was committed to their repository. I asked in the upstream bug which fix is correct and there was no reply so you might at least want to document which of the three solutions (or a fourth one?) you chose any why that's the correct one.

Reply:
@ryandesign Thank you, and sorry that I did not check for existing tickets (and forgot there was mine). My patch looks identical to the one here: https://sourceforge.net/p/freeimage/patches/153
Assuming that original code from upstream was meaningful, this looks correct.

In addition, however, patching `OpenEXR` part is required, but that should be non-controversial, I literally borrowed it from `OpenEXR` master.

Reply:
Take a look here: https://github.com/WinMerge/freeimage/blob/0e3616264c13c713e99e4723acdafa12cadf605b/Source/FreeImage/PluginDDS.cpp#L131-L167 (just for the sake of convenience, we know it is a clone).
And this: https://github.com/WinMerge/freeimage/blob/0e3616264c13c713e99e4723acdafa12cadf605b/Source/FreeImage/PluginDDS.cpp#L254-L255
Those two combined with the error you get trying to build it for BE arch explain my fix here: https://github.com/WinMerge/freeimage/blob/0e3616264c13c713e99e4723acdafa12cadf605b/Source/FreeImage/PluginDDS.cpp#L345-L373 (and yeah, turned out it was suggested much earlier and I should have just checked that ticket).
I think upstream “ported the code to BE” on paper and never bothered to compile it.